backend/app/usecase/validator/alias_test.go

```diff
@@ -42,3 +42,37 @@ func TestCustomAlias_IsValid(t *testing.T) {
 		})
 	}
 }
+
+func TestCustomAlias_HasFragmentCharacter(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name       string
+		alias      string
+		expIsValid bool
+	}{
+		{
+			name:       "empty string",
+			alias:      "",
+			expIsValid: false,
+		},
+		{
+			name:       "alias with fragment",
+			alias:      "fb#home",
+			expIsValid: true,
+		},
+		{
+			name:       "alias without fragment",
+			alias:      "fb",
+			expIsValid: false,
+		},
+	}
+
+	validator := NewCustomAlias()
+	for _, testCase := range testCases {
+		testCase := testCase
+		t.Run(testCase.name, func(t *testing.T) {
+			t.Parallel()
+			assert.Equal(t, testCase.expIsValid, validator.HasFragmentCharacter(testCase.alias))
```

Comment:
Now we can change this to `isValid` and check the violation type!

Reply:
TODO
- [ ] Remove `HasFragmentCharacter` test and add test for `isValid` checking for violation type

Reply:
@Coteh Please note that `HasFragmentCharacter` (public) should be renamed to `hasFragmentCharacter ` (private)